[Leo Zheng Rui, Darren] iP

dsicol opened the pull request.

dsicol committed:
Add greet and exit messages

dsicol committed:
Update variable names

dsicol committed:
Fix bugs

dsicol committed:
Inu class: refactor code for shiba personality

Change from Duke to Inu bot.

Add new greet, echo and exit messages.

- Exit program with case-insensitive input "bye" to allow easier exit

dsicol committed:
Add todo list

dsicol committed:
Feat task marking and unmarking

Add task class

- Help with keeping track of completed tasks

dsicol committed:
Remove listInfo variable

dsicol committed:
Style coding standard

dsicol committed:
Add todo, deadline and event task types

dsicol committed:
Edit Event class toString() method

dsicol committed:
Add Ui and Messages classes and update text-ui-tests

dsicol committed:
Refactor code to adhere to coding standards

YongbinWang commented on src/main/java/Inu.java:
```diff
@@ -0,0 +1,141 @@
+    public static final int INDEX_UNMARK = 7;
+    public static final int INDEX_TODO = 5;
+    public static final int INDEX_DEADLINE = 9;
+    public static final int INDEX_EVENT = 6;
```
I like how you name your constants, it is very clear and concise.

YongbinWang commented on src/main/java/Inu.java:
```diff
@@ -0,0 +1,141 @@
+
+    public String fetchToDo(int indexOfStartOfToDo, String userString) {
+        return userString.substring(indexOfStartOfToDo);
+    }
```
I like how you name your methods and follow pascalcase formatting well.

YongbinWang commented on src/main/java/Inu.java:
```diff
@@ -0,0 +1,141 @@
+
+        line = input.nextLine();
+        checkLine = line.toLowerCase();
+
```
Maybe you can consider following a standard line spacing format such as no spacings inside a {} or one spacing inside a {}.

YongbinWang commented on src/main/java/Inu.java:
```diff
@@ -0,0 +1,141 @@
+        String[] deadLine = new String[2];
+        deadLine[INDEX_TODO_INFORMATION] = userString.substring(indexOfStartOfToDo, indexOfStartOfDeadLine);
+        deadLine[INDEX_DEADLINE_INFORMATION] = userString.substring(indexOfStartOfDeadLine + INDEX_OFFSET);
+        return deadLine;
```
I like that array specifiers must be attached to the type not the variable.

YongbinWang commented on src/main/java/Inu.java:
```diff
@@ -0,0 +1,141 @@
+                unMarkTask(taskIndex);
+                Ui.printMessage(Messages.MESSAGE_UNMARK_TASK + "\n" + taskList.getTask(taskIndex).toString());
+
+            } else if (checkLine.startsWith("todo ")) {
```
Maybe you can consider switching to switch cases for readability.

YongbinWang commented on src/main/java/Inu.java:
```diff
@@ -0,0 +1,141 @@
+                taskList.addTask(eventTask);
+
+                Ui.printMessage("added: " + taskList.getLastTask().toString() + "\n"
+                        + "Now you have " + taskList.getListIndex() + " tasks in your list.");
```
I like that your cases are defined clearly to avoid nesting, it is very clear.

Sherlock-YH commented on src/main/java/Inu.java:
```diff
@@ -0,0 +1,141 @@
+
+    public String[] fetchEvent(
+            int indexOfStartOfToDo, int indexOfStartOfFrom, int indexOfStartOfTo, String userString) {
+        final int INDEX_OFFSET = 1;
+        final int INDEX_TODO_INFORMATION = 0;
+        final int INDEX_EVENT_FROM_INFORMATION = 1;
+        final int INDEX_EVENT_TO_INFORMATION = 2;
```
Maybe can put some simple descriptions about what these integers mean

Sherlock-YH reviewed (commented):
OVERALL GOOD JOB

YongbinWang reviewed (commented):
Looks good to me. Overall code structure is very smooth and following coding convections very well.

dsicol committed:
Add Util class to parse command

dsicol committed:
Refactor Inu and Util classes to utilise switch statements

dsicol committed:
Add UserCommands class to handle inputs and refactor code

dsicol committed:
Fix bugs in UserCommands and update text-ui-tests

dsicol committed:
Add exceptions and ExceptionManager class

This is to handle exceptions with user inputs

dsicol committed:
Improve exception handling and fix bugs

dsicol committed:
Update text-ui-test files to work with exceptions

dsicol committed:
Improve code style

dsicol committed:
Merge branch 'branch-Level-5'

dsicol committed:
Add packages

- Add inu.exception.handling to file exception handling
- Add inu.input.and.output to file user input and output control
- Add inu.task and inu.task.list to maintain Task related Class types

dsicol committed:
Rename packages and update runtest.bat directory

dsicol committed:
Merge branch 'branch-A-Packages'

dsicol committed:
Add delete feature and change task list to ArrayList

- Arraylist better supports resizing

dsicol committed:
Add file saving to save task list

- Saves user task list on different runs of the app

dsicol committed:
Merge branch 'branch-Level-6'

dsicol committed:
Merge branch 'branch-Level-7'

dsicol committed:
Fix bugs to work with ArrayList

dsicol committed:
Create jar file

dsicol committed:
Refactor code in classes and make Task abstract class

dsicol committed:
Improve code style and quality

- Style import messages

dsicol committed:
Add Parser class and refactor code

- Update package contents
- Remove import wildcards
- Improve code standard and quality

cheehengk commented on src/main/java/inu/commons/Util.java:
```diff
@@ -0,0 +1,48 @@
+    public static String fetchFrom(String userString) throws EmptyUserInputException {
+        int firstSlashEntry = userString.indexOf("/");
+        int secondSlashEntry = userString.lastIndexOf("/");
+        if (firstSlashEntry != secondSlashEntry) {
```
Would be better here to use the 'if' condition to check and throw the error so that it is clearer what the error means. The correct/happy path should be prominent.

cheehengk commented on src/main/java/inu/exceptionhandling/ExceptionManager.java:
```diff
@@ -0,0 +1,30 @@
+public class ExceptionManager {
+
+    public static void checkEmptyUserInput(String... strings) throws EmptyUserInputException {
+
+        for (String s : strings) {
+
+            if (s.isEmpty()) {
+
+                throw new EmptyUserInputException();
+
+            }
+        }
+
+    }
```
This is a short method, can do without empty lines since it is readable without.

cheehengk commented on src/main/java/inu/parser/Parser.java:
```diff
@@ -0,0 +1,216 @@
+import inu.commons.Util;
+
+
+public class Parser {
```
Parser class is very long, and seem to do more than just 'parsing'. Consider having command classes if needed.

cheehengk commented on src/main/java/inu/parser/Parser.java:
```diff
@@ -0,0 +1,216 @@
+
+        } catch (IndexOutOfBoundsException e) {
+            Ui.printPromptValidTaskIndex();
+        } catch (EmptyUserInputException e) {
```
See line 130. Be consistent throughout your codebase as to whether there is an empty line on the first line of the catch block. That being said, ensure that the use of them is consistent and not excessive.

cheehengk commented on src/main/java/inu/parser/Parser.java:
```diff
@@ -0,0 +1,216 @@
+
+    protected static final int DECODED_TO_DATE = 4;
+
+    protected static final String T = "T";
+
+    protected static final String D = "D";
+
+    protected static final String E = "E";
```
Constant names are exactly the same as the string assigned to them. Think why we are using constants, perhaps give more meaningful names?

cheehengk commented on src/main/java/inu/parser/Parser.java:
```diff
@@ -0,0 +1,216 @@
+
+        } catch (EmptyUserInputException e) {
+
+            Ui.printPromptValidFromAndToEntryAfterSlash();
```
Avoid using method or variable names which are too long and difficult to read/understand. Be specific but succinct :)

cheehengk reviewed (commented):
Overall very good effort in applying content taught with regards to code quality. Evident work is also done to improve readability. Relatively short methods and clear naming were used.

dsicol committed:
Edit file path and rebuild jar file

dsicol committed:
Refactor code with more OOP

- Add Command abstract class
- Add sub-classes of command

dsicol committed:
Improve code style

- Make happy path prominent
- Remove import wildcards

dsicol committed:
Add date and time feature

- Deadlines and Events to support dates and times
- Input in the form of <DD/MM/YYYY HH:MM> in 24 hour clock
- toString() prints in Day Month Year and Time in 12 hour clock format

dsicol committed:
Add feature to print tasks by date

dsicol committed:
Merge pull request #1 from dsicol/branch-Level-8

Add date and time feature

dsicol committed:
Merge branch 'master' of https://github.com/dsicol/ip

dsicol committed:
Add find feature

- Filters task list by key word

dsicol committed:
Merge pull request #2 from dsicol/branch-Level-9

Add find feature

dsicol committed:
Merge branch 'master' of https://github.com/dsicol/ip

dsicol committed:
Improve UI

dsicol committed:
Refactor code

- Use streams
- Improve code quality for filtering

dsicol committed:
Add JavaDocs

dsicol committed:
Merge pull request #3 from dsicol/branch-A-JavaDoc

Add JavaDocs

dsicol committed:
Merge branch 'master' of https://github.com/dsicol/ip

dsicol committed:
Add User Guide

dsicol committed:
Refactor code and fix bugs
- Add trim method to input

dsicol committed:
Edit Parser class catch clauses to be neat

dsicol committed:
Fix bug where event from can happen after to date

dsicol committed:
Fix bug where past date can be added to task list

dsicol committed:
Add more JavaDocs

dsicol committed:
Add help command

dsicol committed:
Use String formatting

dsicol committed:
Edit JavaDocs comments

dsicol committed:
Add help feature in user guide and rebuild jar

dsicol committed:
Edit feature to save task list after successful commands

dsicol committed:
Edit messages

dsicol committed:
Improve code style

dsicol committed:
Fix bugs

dsicol committed:
Update User Guide

dsicol committed:
Edit messages for empty printed task list

dsicol committed:
Update user guide to reflect v0.21
